packages/packages-groonga-org-package-task.rb

```diff
@@ -175,19 +175,32 @@ def download_packages(target_namespace)
     mkdir_p(repositories_dir)
     download_dir = "#{base_dir}/tmp/downloads/#{@version}"
     mkdir_p(download_dir)
+    
+    github_releases_base_url = "https://github.com/groonga/groonga/releases/download"
+    htaccess_path = "#{repositories_dir}/.htaccess"
+
+    if (target_namespace == :windows)
+      htaccesss_value = "RewriteRule groonga-((1[2-9]|[2-9][0-9])[\\d\\.]+?)-.+?-vs(?!.*(2012|2013|2015|2017)).+?\\.zip"\
+                        " #{github_releases_base_url}/v$1/$0\n"
+      File.write(htaccess_path, htaccesss_value, mode: "w")
```

Comment:
Memo:

When vs2019+ reaches EOL, I think we need to add RewriteRules for the `latest` version of them at this point.

For example, if `vs2019` reaches EOL and `#{target_namespace}_targets` not contains `vs2019`, we don't create RewriteRules for `vs2019` on the lines 191 - 203.
So users will not get latest `vs2019` from URLs like https://packages.groonga.org/windows/groonga/groonga-latest-x86-vs2019-with-vcruntime.zip, they will return 404 not found.
In order to avoid 404 not found, we need to add RewriteRules like below.


```
RewriteRule groonga-latest-x86-vs2019-with-vcruntime.zip https://github.com/groonga/groonga/releases/download/{old-version}/groonga-{old-version}-x86-vs2019-with-vcruntime.zip
```

If we don't need to provide the way to download EOL versions from packages.groonga.org, the RewriteRules for `latest` of EOL versions is needless.

Reply:
この英語では何を言っているのかわからない可能性があるので日本語でも記載。

191行目から203行目では、現在サポート対象となっているVisual Studioのバージョン(vsXXXX)の`latest`に対するリダイレクトルールを作っている。
なので、サポート対象外になったvsXXXXのlatestへのリダイレクトルールは作られない。
毎回新しく.htaccessを作っているので、.htaccessの中に古いバージョン用の`latest`用のリダイレクトルールは作られない。

vs2017まではpackages.groonga.orgに実体があるからそれでも問題ないのだが（.htaccessのルールに引っかからなくても、実体のほうが使われるだけだから）、この対応が入ったvs2019以降では、packages.groonga.orgに実体がない。
そのため、例えばvs2019のサポートが終了して、latest-xxx-vs2019のリダイレクトルールが作られなくなると、 https://packages.groonga.org/windows/groonga/groonga-latest-x86-vs2019-with-vcruntime.zip のようなURLにアクセスした時に、404エラーになる。
なので、このあたりの処理で https://packages.groonga.org/windows/groonga/groonga-latest-x86-vs2019-with-vcruntime.zip が古い（vs2019を最後にサポートした）バージョンへのリダイレクトルールを作ってあげる必要があるのではないかということ。

ただ、サポートが終了したバージョンはpackages.groonga.orgからはダウンロードさせなくても良いというのであれば、そのような対処は不要。
（サポート終了したバージョンはGithubのReleasesからダウンロードしてくれ、というような記載をマニュアルにする必要はあると思うが。）

Reply:
すでにEOLになったものは、サポート終了しているので、 https://packages.groonga.org/windows/groonga/groonga-latest-x86-vs2019-with-vcruntime.zip でダウンロードできなくてもいいように思います。

Reply:
なるほど、であれば特に将来的に追加で対応が必要になることはなさそうですね！